reactiveweb/src/ember-concurrency.ts

```diff
@@ -193,11 +196,22 @@ export class State<Args extends any[], Return, LocalTask extends TaskIsh<Args, R
     return this.lastTask?.value;
   }
 
-  [RUN] = (positional: Args) => {
+  [RUN] = () => {
+    let args = this[THUNK] || DEFAULT_THUNK;
```

Comment:
instead of doing `this[THUNK] || DEFAULT_THUNK` here, could we move it to the constructor.

this way, we can access the args directly in here (still invoking normalizeThunk as you have it though).